File: src/main/java/wordseeker/gui/table/TableModel.java

package wordseeker.gui.table;

import javax.swing.table.AbstractTableModel;
import java.util.ArrayList;
import java.util.List;

/**
 * The class instantiates AbstractTableModel and used as table model in Table.
 *
 * @author rad1kal
 */
public class TableModel extends AbstractTableModel {
    private final List<TableEntry> entryList;
    private static TableModel table;

    /**
     * Construct class instance and initialise entryList.
     */
    private TableModel() {
        entryList = new ArrayList<TableEntry>();
    }

    public static synchronized TableModel getInstance() {
        if (table == null)
            table = new TableModel();
        return table;
    }

    public List<TableEntry> getListOfEntries() {
        return entryList;
    }

    @Override
    public int getRowCount() {
        if (entryList != null)
            return entryList.size();
        else
            return 1;
    }

    @Override
    public int getColumnCount() {
        return 2;
    }

    @Override
    public String getColumnName(int c) {
        switch (c) {
            case 0:
                return "Путь к файлу";
            case 1:
                return "Найденная строка";
            default:
                return "";
        }
    }

    @Override
    public boolean isCellEditable(int row, int col) {
        switch (col) {
            case 0:
                return false;
            case 1:
                return true;
            default:
                return false;
        }
    }

    @Override
    public Object getValueAt(int rowIndex, int columnIndex) {
        if (entryList != null) {
            switch (columnIndex) {
                case 0:
                    return entryList.get(rowIndex).getPathToFile();
                case 1:
                    return entryList.get(rowIndex).getFoundInFile();
            }
        }
        return null;
    }
}

File: src/main/java/wordseeker/gui/table/TableModel.kt
package wordseeker.gui.table

import javax.swing.table.AbstractTableModel
import java.util.ArrayList

/**
 * The class instantiates AbstractTableModel and used as table model in Table.

 * @author rad1kal
 */
class TableModel
/**
 * Construct class instance and initialise entryList.
 */
private constructor() : AbstractTableModel() {
    val listOfEntries: ArrayList<TableEntry>?

    init {
        listOfEntries = ArrayList<TableEntry>()
    }

    override fun getRowCount(): Int {
        if (listOfEntries != null)
            return listOfEntries.size
        else
            return 1
    }

    override fun getColumnCount(): Int {
        return 2
    }

    override fun getColumnName(c: Int): String {
        when (c) {
            0 -> return "Путь к файлу"
            1 -> return "Найденная строка"
            else -> return ""
        }
    }

    override fun isCellEditable(row: Int, col: Int): Boolean {
        when (col) {
            0 -> return false
            1 -> return true
            else -> return false
        }
    }

    override fun getValueAt(rowIndex: Int, columnIndex: Int): Any? {
        if (listOfEntries != null) {
            when (columnIndex) {
                0 -> return listOfEntries[rowIndex].pathToFile
                1 -> return listOfEntries[rowIndex].foundInFile
            }
        }
        return null
    }

    companion object {
        private var table: TableModel = TableModel()

        val instance: TableModel
            @Synchronized get() {
                return table
            }
    }
}
